apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -1205,31 +1205,12 @@ func (m *ModelManagerAPI) getModelInfo(ctx context.Context, tag names.ModelTag,
 		modelAdmin = err == nil
 	}
 
-	users, err := model.Users()
-	if shouldErr(err) {
-		return params.ModelInfo{}, errors.Trace(err)
-	}
-	if err == nil {
-		for _, user := range users {
-			if !modelAdmin && m.authCheck(user.UserTag) != nil {
-				// The authenticated user is neither the controller
-				// superuser, a model administrator, nor the model user, so
-				// has no business knowing about the model user.
-				continue
-			}
-
-			userInfo, err := common.ModelUserInfo(user, model)
-			if err != nil {
-				return params.ModelInfo{}, errors.Trace(err)
-			}
-			info.Users = append(info.Users, userInfo)
-		}
-
-		if len(info.Users) == 0 {
-			// No users, which means the authenticated user doesn't
-			// have access to the model.
-			return params.ModelInfo{}, errors.Trace(apiservererrors.ErrPerm)
-		}
+	info.Users, err = common.ModelUserInfo(ctx, m.accessService, m.apiUser, tag)
+	if errors.Is(err, accesserrors.PermissionNotValid) {
+		return params.ModelInfo{}, errors.Annotate(apiservererrors.ErrPerm, "getting model user info")
+	} else if err != nil &&
+		!(errors.Is(err, modelerrors.NotFound) && ignoreNotFoundError) {
+		return params.ModelInfo{}, errors.Annotate(err, "getting model user info")
```

Comment:
Can we simplify this, the `ignoreNotFoundError` isn't helping here.

Also add a comment on why we can ignore the NotFound error, why is it safe to do so?

Reply:
Changed to use the local `shouldErr` function used elsewhere in this method and updated it to also check for modelNotFound. It already has documentation on so hopefully that should suffice.